python_utils/rec.py

```diff
@@ -0,0 +1,95 @@
+import sys
+import matplotlib.pyplot as plt
+import math
+import numpy as np
+
+perm_amp = []
+perm_phase = []
+# Create figure for plotting
+plt.ion()
+fig = plt.figure()
+ax = fig.add_subplot(111)
+count = 0
+plt.xlabel("X-axis")
+plt.ylabel("Y-axis")
+plt.title("Updating plot...")
+fig.canvas.draw()
+plt.show(block=False)
+process_line = True
+
+
+def carrier_plot(amp):
+
+    #Adds xlims after 100 entries
+    if count > 100:
+        plt.xticks([], [])
+        plt.xlim()
+
+    else:
+        plt.xlim(0, 100)
+    plt.xlabel("Packet number/Time")
+    plt.ylabel("Amplitude")
+    plt.rcParams["figure.figsize"] = [7.50, 3.50]
+    plt.rcParams["figure.autolayout"] = True
+    plt.title("Line graph")
+    plt.show()
+
+    df = np.asarray(amp)
+    # print(df.shape)
+    # Can be changed to df[x] to plot sub-carrier x only (set color='r' also)
+    plt.plot(df[:, 44], color='r')
+    # fig.canvas.draw()
+
+    # TODO use blit instead of flush_events for more fastness
+    # to flush the GUI events
+    fig.canvas.flush_events()
+    plt.clf()
+
+
+def process(res):
+
+    # Parser
+    all_data = res.split(',')
+    csi_data = all_data[25].split(" ")
+    csi_data[0] = csi_data[0].replace("[", "")
+    csi_data[-1] = csi_data[-1].replace("]", "")
+
+    csi_data.pop()

```

Comment:
This will not be required with `deque` (see below). I believe pop is inefficient (see: https://stackoverflow.com/questions/4151320/efficient-circular-buffer)

Reply:
Pop is used to remove the header. I dont understand why deque fixes that. I have tried implementing deque but having trouble.

Reply:
Sorry, I misunderstood the purpose of this line. However, I do not think this pops the header. I think this removes the final element in `csi_data` which I assume is just an empty string. In which case, this implies that line 51 is not necessary. 

Either way, I think we can just leave this as-is for now.